### DISCUSSION TITLE
How to create a new migration offline

### DISCUSSION THREAD

**Discussion by ghost** [prisma/prisma#11541] [Q&A] [answered]
I'm new to prisma, so please forgive me as I may have misunderstood the concept of prisma.

I feel strange about the behavior of the --create-only option of the migration dev command.
https://www.prisma.io/docs/reference/api-reference/command-reference

--create-only option, as the name implies, feels like it doesn't require a connection to the DB if it just creates a new migration based on a schema change, but it's actually a DB Connection to is required.

Could Prisma create a new migration offline?

**Comment by austincrim** [comment, marked as answer]
Hey @yogarasu 👋 ,

When running `prisma migrate dev`, Prisma uses a ["shadow database"](https://www.prisma.io/docs/concepts/components/prisma-migrate/shadow-database) behind-the-scenes in order to detect schema drift and provide relevant error messages. This requires a connection to your database.

Let me know if I can clarify anything!
